fe/fe-core/src/main/java/com/starrocks/connector/iceberg/io/IcebergCachingFileIO.java

```diff
@@ -248,18 +308,30 @@ private TwoLevelContentCache() {
 
             Caffeine<Object, Object> diskCacheBuilder = Caffeine.newBuilder();
             this.diskCache = diskCacheBuilder.maximumWeight(DISK_CACHE_CAPACITY)
-                    .weigher((Weigher<String, DiskCacheEntry>) (key, value) -> (int) Math.min(value.length, Integer.MAX_VALUE))
+                    .expireAfterAccess(DISK_CACHE_EXPIRATION_SECONDS, TimeUnit.SECONDS)
+                    .weigher((Weigher<String, DiskCacheEntry>) (key, value) ->
+                            value.useCount == 0 ? (int) Math.min(value.length, Integer.MAX_VALUE) : 0)
                     .recordStats()
-                    .removalListener(((key, value, cause) -> {
-                        LOG.debug(key + " to be eliminated from disk, reason: " + cause);
-                        HadoopOutputFile hadoopOutputFile = (HadoopOutputFile) IOUtil.getOutputFile(
-                                METADATA_CACHE_DISK_PATH, key);
-                        try {
-                            hadoopOutputFile.getFileSystem().delete(hadoopOutputFile.getPath(), false);
-                        } catch (Exception e) {
-                            LOG.warn("failed on deleting file :" + hadoopOutputFile.getPath());
+                    // use sync CacheWriter to avoid delete file newly generated by another thread
+                    .writer(new CacheWriter<String, DiskCacheEntry>() {
+                        @Override
+                        public void write(@NonNull String s, @NonNull DiskCacheEntry diskCacheEntry) {
```

Comment:
do not need to impl write?

Reply:
cache will call write when create an entry, we don't need do more thing when insert or update an entry, so it's empty.